### PR TITLE
COMDOX-284: Initial repo setup

### PR DESCRIPTION
**Pull request by jeff-matthews**
## Description

This PR modifies the files provisioned by the default devsite doc template in preparation for migrating content from `devdocs.magento.com` in the following ways:

- Add Yarn support for package management
- Update validation workflow to include internal link checking and test builds
- Update gh-pages workflow to be on demand instead of on push to main
- Add modular navigation
- Replace Analytics placeholder files with stubs for Commerce Testing
- Add custom rule config for .md linting
- Update files to match other commerce-* repos
   - PR template
   - `.env` sample
   - `.gitignore`
- Add link checker packages

## Related Issue

COMDOX-284

## Motivation and Context

All commerce-* repos should be consistent.

## How Has This Been Tested?

Validation run via GitHub Actions

## Screenshots (if appropriate):

N/A

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by jeff-matthews** [comment]
@bdenham, I ported the changes you made in the `commerce-frontend` repo.

File: package.json [hunk 1]
```diff
@@ -1,21 +1,20 @@
 {
-  "private": true,
-  "name": "dev-site-documentation-template",
+  "name": "commerce-testing",
```
Comment (bdenham) [inline package.json:2]:
```suggestion
  "name": "commerce-admin-developer",
```

File: package.json [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
   "version": "1.0.0",
   "license": "Apache-2.0",
   "repository": {
     "type": "git",
-    "url": "https://github.com/AdobeDocs/dev-site-documentation-template"
-  },
-  "author": {
-    "name": "Stephan Ringel",
-    "url": "https://github.com/icaraps"
+    "url": "https://github.com/AdobeDocs/commerce-testing"
```
Comment (bdenham) [inline package.json:7]:
```suggestion
    "url": "https://github.com/AdobeDocs/commerce-admin-developer"
```

**Review by bdenham — APPROVED** [review]
Looks good from a workflow build/infrastructure POV. Just a couple of cut-paste errors. 👍

**Pull request merged**